### PR TITLE
add dataPath in Cast toUplcData() call tree

### PR DESCRIPTION
**Pull request by rjharmon**
- accumulates hints to where the data is wrong, in case of failure

### CODE REVIEW

File: src/cast/Cast.js
Comment (christianschmitz) [inline src/cast/Cast.js]: I'm a bit confused here, should we clean up `debugger` and `@ts-expect-error`? 
And `ft.tag` will always be `undefined` for `map` I think, only enum variants have tags (the constr index), so `ft.name` or `key` should suffice

File: src/cast/Cast.js
Comment (christianschmitz) [inline src/cast/Cast.js:107]: I'm OK with being able to pass in `dataPath` as an argument here, but could you add a note about the use-case?

**Review by christianschmitz — CHANGES_REQUESTED** [review]
Thanks for this contribution, a little cleanup and an additional comment and it should be good to merge.

Do you think there will be any benefit to also doing this on the `fromUplcData`-side?

File: src/cast/Cast.js
Comment (rjharmon) [reply src/cast/Cast.js]: Yes, I'll get rid of the debugger before merging.

Ex:

```
struct cip68mapStruct {
  fname1: String
  taggedFname: Int "fn2"
}
```

I hadn't determined how to access both the defined field name (`taggedFname`) as well as the serialization format key (`fn2`).  AFAICT the previous version has only the serialization keys and doesn't allow `{fname1, taggedFname}` data.  I'd imagine `{fname1, fn}` would be allowable too (concerns welcome).

I've been a bit blocked on reaching this `debugger` statement ... will do.

File: src/cast/Cast.js
Comment (rjharmon) [reply src/cast/Cast.js:107]: yup, will do

File: src/cast/Cast.js
Comment (christianschmitz) [reply src/cast/Cast.js]: Currently only the tags can be accessed externally, and for fields without explicit tags the field name is used as an implicit tag: 
So here it would be: `{"fname1", "fn2"}`

**Review by christianschmitz — APPROVED** [review]
great work!

**Pull request merged**
